### PR TITLE
effect systems now check if they've been deleted before stepping

### PR DESCRIPTION
**Pull request by optimumtact**
this stops some runtimes

### CODE REVIEW

File: code/game/objects/effects/effect_system/effect_system.dm [hunk 1]
```diff
@@ -69,6 +69,8 @@ would spawn and follow the beaker, even if it is carried or thrown.
 	var/steps_amt = pick(1,2,3)
 	for(var/j in 1 to steps_amt)
 		sleep(5)
+		if(QDELETED(src) || QDELETED(E))
+			return
```
Comment (AnturK) [inline code/game/objects/effects/effect_system/effect_system.dm]: In case of E won't this leave undeleted system around (see autocleanup below)

Comment (optimumtact) [reply code/game/objects/effects/effect_system/effect_system.dm]: it's to guard against one of many effect types going away?

Comment (optimumtact) [reply code/game/objects/effects/effect_system/effect_system.dm]: oh I think I misunderstood your comment, are you wanting the total effects counter decremented?

Comment (optimumtact) [reply code/game/objects/effects/effect_system/effect_system.dm]: or just a qdelete of the entire effects system?

Comment (AnturK) [reply code/game/objects/effects/effect_system/effect_system.dm]: Yeah, if only one of the effects get nicked somehow the whole system could possibly stay around when it shouldn't.

**Pull request closed without merging**